include/llama/Copy.hpp

```diff
@@ -0,0 +1,364 @@
+// SPDX-License-Identifier: GPL-3.0-or-later
+
+#pragma once
+
+#include "View.hpp"
+#include "mapping/AoSoA.hpp"
+#include "mapping/SoA.hpp"
+
+#include <cstring>
+#include <numeric>
+
+namespace llama
+{
+    namespace internal
+    {
+        inline void parallel_memcpy(
+            std::byte* dst,
+            const std::byte* src,
+            std::size_t size,
+            std::size_t threadId = 0,
+            std::size_t threadCount = 1,
+            decltype(std::memcpy) memcpy = std::memcpy)
+        {
+            const auto sizePerThread = size / threadCount;
+            const auto sizeLastThread = sizePerThread + size % threadCount;
+            const auto sizeThisThread = threadId == threadCount - 1 ? sizeLastThread : sizePerThread;
+            memcpy(dst + threadId * sizePerThread, src + threadId * sizePerThread, sizeThisThread);
```

Comment:
Isn't this a possible shadowing of C's `memcpy`?

Reply:
Technically, C++ compilers should not provide `memcpy` in the global namespace. Practically, they often do. So yes, this is a shadowing. I will rename it.